independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanArchives.java

```diff
@@ -209,14 +221,28 @@ private void getKnownImplementors(DotName name, Set<ClassInfo> allKnown, Set<Dot
             }
         }
 
+        private Optional<ClassInfo> computeAdditional(DotName className) {
+            LOGGER.debugf("Index: %s", className);
+            Indexer indexer = new Indexer();
+            if (BeanArchives.index(indexer, className.toString())) {
+                Index index = indexer.complete();
+                return Optional.of(index.getClassByName(className));
+            } else {
+                // Note that ConcurrentHashMap does not allow null to be used as a value
+                return Optional.empty();
+            }
+        }
+
     }
 
-    static void index(Indexer indexer, String className) {
+    static boolean index(Indexer indexer, String className) {
         try (InputStream stream = BeanProcessor.class.getClassLoader()
                 .getResourceAsStream(className.replace('.', '/') + ".class")) {
             indexer.index(stream);
+            return true;
         } catch (IOException e) {
-            throw new IllegalStateException("Failed to index: " + className, e);
+            LOGGER.warnf("Failed to index %s: %s", className, e.getMessage());
```

Comment:
Could we filter the primitive types somewhere (and the arrays of primitive types)? Thinking about https://github.com/quarkusio/quarkus/issues/3728#issuecomment-525609336 and the initial issue (https://github.com/quarkusio/quarkus/issues/3537).

Because I don't think it's worth throwing a warning for these cases.

Reply:
Well, it's just one line in the log... and TBH I don't think an `IndexView` "client" should attempt to obtain a `ClassInfo` for primitives.

Reply:
Mkay. We'll see how more reports we have for that then :).